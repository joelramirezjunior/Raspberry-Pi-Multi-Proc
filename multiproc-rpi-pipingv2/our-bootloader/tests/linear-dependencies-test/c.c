#include "rpi.h"

int do_stuff(int a) {
    unsigned long long i = 0;
    printk("     doing stuff... should take a bit\n");
    while (0 != i++) {
    	a++;
    }
    return 140 * a + 0xe;
}

void notmain(void) {
    printk("continuing linear dependencies test.\n");
    printk("starting c, which is depended on by d. ----------------------\n");
    //todo: int c = READ FROM PIPE;
    int c = 10;
    printk("doing stuff with starting number of: ---%d---, writing the output so that d can read it\n", c);
    c = do_stuff(c);
    trace("%d", c);
    printk("wrote output of ---%d--- which d will expect\n", c);
    printk("done with c. moving on... --------------------------------------\n");
}

