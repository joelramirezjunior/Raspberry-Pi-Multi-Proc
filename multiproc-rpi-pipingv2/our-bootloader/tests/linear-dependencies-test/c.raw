FOUND: </dev/cu.SLAB_USBtoUART>
opened tty port </dev/cu.SLAB_USBtoUART>.
pi-install: tty-usb=</dev/cu.SLAB_USBtoUART> program=<./linear-dependencies-test/c.bin> about to boot
TRACE:simple_boot: sending 3076 bytes, crc32=b9fec172
waiting for a start
putting code
pi sent print: <UART Teddy Zhang: success: Received the program!>
bootloader: Done.
listening on ttyusb=</dev/cu.SLAB_USBtoUART>
continuing linear dependencies test.
starting c, which is depended on by d. ----------------------
doing stuff with starting number of: ---10---, writing the output so that d can read it
     doing stuff... should take a bit
TRACE:notmain:1414wrote output of ---1414--- which d will expect
done with c. moving on... --------------------------------------
DONE!!!

Saw done

bootloader: pi exited.  cleaning up
