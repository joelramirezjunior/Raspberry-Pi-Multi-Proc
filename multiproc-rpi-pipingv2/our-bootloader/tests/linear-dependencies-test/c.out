TRACE:notmain:1414wrote output of ---1414--- which d will expect
